TheSadRogue.Primitives/PolarCoordinate.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace SadRogue.Primitives
+{
+    /// <summary>
+    /// A Polar Coordinate.
+    /// </summary>
+    /// <remarks>
+    /// See wikipedia: https://en.wikipedia.org/wiki/Polar_coordinate_system
+    /// Polar Coordinates are very, very slow and should not be used often
+    /// </remarks>
+    public class PolarCoordinate
+    {
+        /// <summary>
+        /// The distance away from the Origin (0,0) of this Polar Coord
+        /// </summary>
+        public readonly double Radius;
+
+        /// <summary>
+        /// The degree of rotation, clockwise
+        /// </summary>
+        public readonly double Theta; //in degrees clockwise
+
+        /// <summary>
+        /// Creates a new Polaar Coordinate with the given Radius and Theta
+        /// </summary>
+        /// <param name="radius">Radius of the Polar Coord</param>
+        /// <param name="theta">Degree of rotation (clockwise) of the Polar Coord</param>
+        public PolarCoordinate(double radius, double theta)
+        {
+            Radius = radius;
+            Theta = theta;
+        }
+
+        public static bool operator ==(PolarCoordinate left, PolarCoordinate right)
+        {
+            if (left.Theta > right.Theta - 0.05 && left.Theta < right.Theta + 0.05)
+                if (left.Radius > right.Radius - 0.05 && left.Radius < right.Radius + 0.05)
+                    return true;
+
+            return false;
+        }
+
+        public static bool operator !=(PolarCoordinate left, PolarCoordinate right)
+        {
+            return !(left == right);
+        }
+
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
+        }
+        public override bool Equals(object obj)
+        {
+            if (obj.GetType() != typeof(PolarCoordinate))
+                return false;
+            else
+                return this == (PolarCoordinate)obj;
+        }
+
+        /// <summary>
+        /// A handly list of polar functions for you.
+        /// </summary>
+        /// <remarks>
+        /// Each function accepts a parameter of Theta, which is the angle of rotation around the center.
+        /// They return a Point that can be used on the map.
+        /// Polar functions are very slow. Use them seldom, for things like spirographs
+        /// As the radius gets large, there will be increased likelihood
+        /// of functions producing gaps in the output.
+        /// You've been warned.
+        /// </remarks>
+        public static List<Func<double, Point>> PolarFunctions => new List<Func<double, Point>>

```

Comment:
Nice!  Are these just examples, or functions intended for a user to grab out and use?  If they're intended for the user, is there a way to make them more accessible?  Right now they'd have to look at the source and grab an index out.  Perhaps `Dictionary<string, Func<double, Point>>`?